test/parallel/test-net-connect-options-ipv6.js

```diff
@@ -27,63 +29,48 @@ if (!common.hasIPv6)
 const net = require('net');
 
-const hosts = common.localIPv6Hosts;
-let hostIdx = 0;
-let host = hosts[hostIdx];
-let localhostTries = 10;
+const hostAddrIPv6 = '::1';
+const HOSTNAME = 'dummy';
 
-const server = net.createServer({ allowHalfOpen: true }, function(socket) {
+const server = net.createServer({ allowHalfOpen: true }, (socket) => {
   socket.resume();
   socket.on('end', common.mustCall());
   socket.end();
 });
 
-server.listen(0, '::1', tryConnect);
-
 function tryConnect() {
-  const client = net.connect({
-    host: host,
+  const connectOpt = {
+    host: HOSTNAME,
     port: server.address().port,
     family: 6,
-    allowHalfOpen: true
-  }, function() {
-    console.error('client connect cb');
+    allowHalfOpen: true,
+    lookup: common.mustCall((addr, opt, cb) => {
+      assert.strictEqual(addr, HOSTNAME);
+      assert.strictEqual(opt.family, 6);
+      cb(null, hostAddrIPv6, opt.family);
+    })
+  };
+  // No `mustCall`, since test could skip, and it's the only path to `close`.
+  const client = net.connect(connectOpt, () => {
     client.resume();
-    client.on('end', common.mustCall(function() {
+    client.on('end', () => {
+      // Wait for next uv tick and make sure the socket stream is writable.
       setTimeout(function() {
         assert(client.writable);
         client.end();
       }, 10);
-    }));
-    client.on('close', function() {
-      server.close();
     });
-  }).on('error', function(err) {
-    // ENOTFOUND means we don't have the requested address. In this
-    // case we try the next one in the list and if we run out of
-    // candidates we assume IPv6 is not supported on the
-    // machine and skip the test.
-    // EAI_AGAIN means we tried to remotely resolve the address and
-    // timed out or hit some intermittent connectivity issue with the
-    // dns server.  Although we are looking for local loopback addresses
-    // we may go remote since the list we search includes addresses that
-    // cover more than is available on any one distribution. The
-    // net is that if we get an EAI_AGAIN we were looking for an
-    // address which does not exist in this distribution so the error
-    // is not significant and we should just move on and try the
-    // next address in the list.
-    if ((err.syscall === 'getaddrinfo') && ((err.code === 'ENOTFOUND') ||
-                                            (err.code === 'EAI_AGAIN'))) {
-      if (host !== 'localhost' || --localhostTries === 0)
-        host = hosts[++hostIdx];
-      if (host)
-        tryConnect();
-      else {
-        server.close();
-        common.skip('no IPv6 localhost support');
-      }
-      return;
+    client.on('close', () => server.close());
+  });
+  client.on('error', (err) => {
+    // ENOTFOUND means we don't have the requested address, so we assume IPv6
+    // is not supported on the machine and skip the test.
+    if ((err.syscall === 'getaddrinfo') && (err.code === 'ENOTFOUND')) {
```

Comment:
I don't think we need this anymore if we are mocking `lookup`? AFAICT the errors are emitted there.

Reply:
I'll remove and stress test.